src/Symfony/Bridge/Twig/Tests/Extension/HttpFoundationExtensionTest.php

```diff
@@ -41,6 +41,14 @@ public function getGenerateAbsoluteUrlData()
             array('http://example.com/baz', 'http://example.com/baz', '/'),
             array('https://example.com/baz', 'https://example.com/baz', '/'),
             array('//example.com/baz', '//example.com/baz', '/'),
+
+            array('http://localhost/foo/bar?baz', '?baz', '/foo/bar'),
+            array('http://localhost/foo/bar?baz=1', '?baz=1', '/foo/bar?foo=1'),
+            array('http://localhost/foo/baz?baz=1', 'baz?baz=1', '/foo/bar?foo=1'),
+
+            array('http://localhost/foo/bar#baz', '#baz', '/foo/bar'),
+            array('http://localhost/foo/bar?baz=1#baz', '?baz=1#baz', '/foo/bar?foo=1'),
+            array('http://localhost/foo/baz?baz=1#baz', 'baz?baz=1#baz', '/foo/bar?foo=1'),
         );
```

Comment:
can you add this test case?

```php
array('http://localhost/foo/bar?0#foo', '#foo', '/foo/bar?0'),
```

I think your current solution will strip out the existing `?0` query string. I know its an edge case but still :smile: